### PR TITLE
cmake: xtensa: use SoC name in path to toolchain

### PR DESCRIPTION
**Pull request by dcpleung**
Xtensa requires building a new toolchain for a specific SoC. By default xtools built Xtensa toolchains all have prefix of xtensa-zephyr-elf. In order to distinguish different toolchains, they are now placed in their own directories under their SoC name. This allows us to have multiple Xtensa toolchains targeting multiple SoCs.

This is in preparation for introducing additional Xtensa toolchains. Requires https://github.com/zephyrproject-rtos/sdk-ng/pull/111.

Requires SDK v0.11.

### CODE REVIEW

File: cmake/toolchain/zephyr/0.11/target.cmake [hunk 1]
```diff
@@ -1,36 +1,32 @@
...
     )
 
   LIST(APPEND TOOLCHAIN_LIBS hal)
   LIST(APPEND LIB_INCLUDE_DIR -L${SYSROOT_DIR}/lib)
+
+  set(CROSS_COMPILE ${TOOLCHAIN_HOME}/xtensa/${SOC_NAME}/${CROSS_COMPILE_TARGET}/bin/${CROSS_COMPILE_TARGET}-)
 elseif("${ARCH}" STREQUAL "x86_64")
   list(APPEND TOOLCHAIN_C_FLAGS -mx32)
   list(APPEND TOOLCHAIN_LD_FLAGS -mx32)
 endif()
```
Comment (SebastianBoe) [inline cmake/toolchain/zephyr/0.11/target.cmake]: Please treat these variables as immutable, e.g.

```
if(ARCH STREQUAL xtensa)
set(CROSS_COMPILE bla bla)
else()
set(CROSS_COMPILE yada yada)
endif()

```

It is easier to debug and read when the variables are not variable.

**Review by SebastianBoe — CHANGES_REQUESTED** [review]

File: cmake/toolchain/zephyr/0.11/target.cmake (hunk 1 above)
Comment (dcpleung) [reply cmake/toolchain/zephyr/0.11/target.cmake]: I have updated the file. Is this what you want?

**Review by SebastianBoe — APPROVED** [review]

**Review by tyrantlizard77 — APPROVED** [review]

**Comment by dcpleung** [comment]
@jhedberg

**Comment by dcpleung** [comment]
Rolled into https://github.com/zephyrproject-rtos/zephyr/pull/20221

**Pull request closed without merging**